### ISSUE TITLE
[Date Picker], [Range Date Picker]: A11y - There is no error message for typing date in unsupported format

### ISSUE THREAD

**Issue by mari-shakhanava** [epam/UUI#2784]
## Description

When User types date in unsupported format, there is no error message, just the field is cleared automatically

## Steps to Reproduce

1. Open [Date Picker](https://uui.epam.com/documents?id=datePicker&mode=doc&isSkin=true&category=components&theme=loveship) component or [Range Date Picker](https://uui.epam.com/documents?id=rangeDatePicker&mode=doc&isSkin=true&category=components&theme=loveship) component
2. Type date in unsupported format (e.g., "14.02.25") and observe the behavior.

## Actual result
https://github.com/user-attachments/assets/ce0bab8a-23ae-453a-ac72-126e0d5df829
There is no error message, just the field is cleared automatically.

## Expected result
Error message should be displayed.

Failed SC:
[3.3.1 Error Identification (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/error-identification)

**Comment by AlekseyManetov** [comment]
Overall, this looks a valid request. However, at the moment, we do not see a good solution to this issue. Additionally, we have not found strong references on how other libraries or major products handle this. Most either prevent keyboard input entirely or reset invalid values in a similar way.

Possible Solutions:

- Modify the API to allow invalid `props.value` – This would introduce a significant breaking change, causing issues for existing projects. They would need to implement their own handling for invalid values.
- Handle invalid values within the input but return null externally – The challenge here is how to visually indicate an invalid state. Component itself can only modify the border color. And users would not be able to attach specific validation messages for this case, since user would't know about invalid state, because of `props.value === null`.
